workflows/prognostic_c48_run/runtime/transformers/fv3fit.py

```diff
@@ -1,101 +1,106 @@
-    def input_variables(self) -> Iterable[Hashable]:
-        nested_vars = [model.input_variables for model in self.models]
-        return list({var for model_vars in nested_vars for var in model_vars})
+        if len(set(states).intersection(tendencies)) > 0:
+            raise ValueError(
+                "A variable cannot be updated by tendency and state predictions."
+            )
+        self.targets = targets
 
-    def predict(self, arg: xr.Dataset) -> Tuple[xr.Dataset, xr.Dataset]:
-        """Predict tendencies and state updates.
+    def invert(self, predictions: State) -> State:
+        """Separate a dataset of predictions into tendency and state predictions.
 
         Args:
-            arg: dataset with input variables.
+            predictions: tendency and/or state predictions, keyed on ML output names.
 
         Returns:
-            Tuple of two datasets. The first dataset contains the tendencies, the
-            second dataset contains the state updates. The keys of each dataset
-            are the state names to be updated.
+            Mapping of tendencies and state updates keyed on state names to be updated.
 
         Note:
-            Tendencies are summed over all predictions for a given variable. For
-            example, if self.tendency_predictions = {'Q1': 'air_temperature',
-            'air_temperature_tendency_due_to_nudging': 'air_temperature'} then
-            the returned tendency dataset will contain a tendency for
-            'air_temperature' that is the sum of these two predicted tendencies.
+            Tendencies are summed over all predictions for a given variable.
         """
-        predictions = []
-        for model in self.models:
-            predictions.append(model.predict(arg))
-        merged_predictions = xr.merge(predictions)
-        tendencies: MutableMapping[str, xr.DataArray] = {}
-        state_updates: MutableMapping[str, xr.DataArray] = {}
-        for tendency_name, variable_name in self.tendency_predictions.items():
-            if variable_name in tendencies:
-                tendencies[variable_name] += merged_predictions[tendency_name]
-            else:
-                tendencies[variable_name] = merged_predictions[tendency_name]
-        for prediction_name, variable_name in self.state_predictions.items():
-            state_updates[variable_name] = merged_predictions[prediction_name]
-        return xr.Dataset(tendencies), xr.Dataset(state_updates)
+        output: MutableMapping[Hashable, xr.DataArray] = {}
+        for ml_output_name, target in self.targets.items():
+            if target.method == "tendency":
+                if target.target_name in output:
+                    output[target.target_name] += predictions[ml_output_name]
+                else:
+                    output[target.target_name] = predictions[ml_output_name]
+            elif target.method == "state":
+                output[target.target_name] = predictions[ml_output_name]
```

Comment:
There is some strong coupling here. Inverter should not know `tendency.method` it should ask MLOutputApplier for all the information it needs. Simple to fix this by refactoring this to a method of `MLOutputApplier` e.g. `get_target_name`. Once this refactor is done, it's probably possible to delete `PredictorInverter` it's method are basically one-liners at that point.